pkg/auth/dependency/identity/anonymous/provider.go

```diff
@@ -66,80 +80,102 @@ func (p *Provider) Delete(i *Identity) error {
 	var key jwk.Key
-	keyFunc := func(token *jwt.Token) (interface{}, error) {
-		// Provided key material has higher priority than key ID
-		if jwkMap, ok := token.Header["jwk"].(map[string]interface{}); ok {
-			var err error
-
-			jwkBytes, err := json.Marshal(jwkMap)
-			if err != nil {
-				return nil, fmt.Errorf("invalid JWK: %w", err)
-			}
-			keySet, err := jwk.ParseBytes(jwkBytes)
-			if err != nil {
-				return nil, fmt.Errorf("invalid JWK: %w", err)
-			}
 
-			key = keySet.Keys[0]
+	hdr, jwtToken, err := jwtutil.SplitWithoutVerify(compact)
+	if err != nil {
+		err = fmt.Errorf("invalid JWT: %w", err)
+		return
+	}
 
-			iden, err = p.Store.GetByKeyID(key.KeyID())
-			if err != nil && !errors.Is(err, identity.ErrIdentityNotFound) {
-				return nil, err
-			} else if err == nil {
-				if key, err = iden.toJWK(); err != nil {
-					return nil, fmt.Errorf("invalid JWK: %w", err)
-				}
-			}
+	err = jwt.Verify(jwtToken,
+		jwt.WithClock(jwtClock{p.Clock}),
+		jwt.WithAcceptableSkew(5*time.Minute),
+	)
+	if err != nil {
+		err = fmt.Errorf("invalid JWT: %w", err)
+		return
+	}
 
-			var ptrKey interface{}
-			err = key.Raw(&ptrKey)
-			if err != nil {
-				return nil, fmt.Errorf("failed to extract key: %w", err)
-			}
+	if jwkIface, ok := hdr.Get("jwk"); ok {
+		var jwkBytes []byte
+		jwkBytes, err = json.Marshal(jwkIface)
+		if err != nil {
+			err = fmt.Errorf("invalid JWK: %w", err)
+			return
+		}
 
-			return ptrKey, nil
+		var set *jwk.Set
+		set, err = jwk.ParseBytes(jwkBytes)
+		if err != nil {
+			err = fmt.Errorf("invalid JWK: %w", err)
+			return
 		}
-		if kid, ok := token.Header["kid"].(string); ok {
-			var err error
-			iden, err = p.Store.GetByKeyID(kid)
-			if err != nil {
-				return nil, fmt.Errorf("unknown key ID: %w", err)
-			}
 
-			if key, err = iden.toJWK(); err != nil {
-				return nil, fmt.Errorf("invalid JWK: %w", err)
-			}
+		key = set.Keys[0]
 
-			var ptrKey interface{}
-			err = key.Raw(&ptrKey)
+		iden, err = p.Store.GetByKeyID(key.KeyID())
+		if err != nil && !errors.Is(err, identity.ErrIdentityNotFound) {
+			return
+		} else if err == nil {
+			key, err = iden.toJWK()
 			if err != nil {
-				return nil, fmt.Errorf("failed to extract key: %w", err)
+				err = fmt.Errorf("invalid JWK: %w", err)
+				return
 			}
-
-			return ptrKey, nil
+		}
+	} else if kid := hdr.KeyID(); kid != "" {
+		iden, err = p.Store.GetByKeyID(kid)
+		if err != nil {
+			err = fmt.Errorf("unknown key ID: %w", err)
+			return
 		}
 
-		return nil, errors.New("no key provided")
+		key, err = iden.toJWK()
+		if err != nil {
+			err = fmt.Errorf("invalid JWK: %w", err)
+			return
+		}
+	} else {
+		err = errors.New("no key provided")
+		return
 	}
 
-	req := &Request{}
-	token, err := jwt.ParseWithClaims(requestJWT, req, keyFunc)
-	if err != nil {
-		return nil, nil, fmt.Errorf("invalid JWT signature: %w", err)
+	// The client does include alg in the JWK.
+	// Fix it by copying alg in the header.
+	if key.Algorithm() == "" {
```

Comment:
This should not happen if our SDK is used?

Reply:
https://github.com/authgear/authgear-sdk-js/blob/master/packages/skygear-react-native/ios/SGSkygearReactNative.m#L417

From my testing, the SDK does not include `alg` in the `jwk`.